libs/vkd3d/resource.c

```diff
@@ -2835,6 +2854,9 @@ static void vkd3d_create_buffer_srv(struct d3d12_desc *descriptor,
     descriptor->magic = VKD3D_DESCRIPTOR_MAGIC_SRV;
     descriptor->vk_descriptor_type = VK_DESCRIPTOR_TYPE_UNIFORM_TEXEL_BUFFER;
     descriptor->u.view = view;
+
+    descriptor->u.view->weak = resource->weak;
+    InterlockedIncrement(&descriptor->u.view->refcount);
```

Comment:
I'd prefer this to happen inside `vkd3d_create_buffer_view_for_resource` and `vkd3d_create_texture_view`, respectively (the latter might need an - optional - resource parameter, or better yet, introduce `vkd3d_create_texture_view_for_resource` or something like that).